test/tools/verify/cmd/main.go

```diff
@@ -51,19 +57,64 @@ func main() {
 	// prepare the manifest
 	manifest, err := readAndModifyManifest(manifestPath, keyValuePairs)
 	if err != nil {
-		printAndDie(fmt.Sprintf("failed to prepare the manifes, due to = %v", err))
+		printAndDie(fmt.Sprintf("failed to prepare the manifest, due to = %v", err))
 	}
 
 	// act
 	err = verify(manifest, kubeClient, machineClient)
 	if err != nil {
-		printAndDie(fmt.Sprintf("failed to verify if a machine/node has been craeted, details = %v", err))
+		printAndDie(fmt.Sprintf("failed to verify if a machine/node has been created, reason: \n%v", err))
 	}
 	fmt.Println("all good, successfully verified that a machine/node has been created within the cluster")
 }
 
 func verify(manifest string, kubeClient kubernetes.Interface, machineClient machineclientset.Interface) error {
-	return fmt.Errorf("not implemented")
+	newMachine := &machinev1alpha1.Machine{}
+	{
+		manifestReader := strings.NewReader(manifest)
+		manifestDecoder := yaml.NewYAMLToJSONDecoder(manifestReader)
+		err := manifestDecoder.Decode(newMachine)
+		if err != nil {
+			return err
+		}
+	}
+
+	// we expect to find no nodes within the cluster
+	err := assureNodeCount(0, kubeClient)
+	if err != nil {
+		return fmt.Errorf("unable to perform the verification, incorrect cluster state detected %v", err)
+	}
+
+	fmt.Printf("creating a new \"%s\" machine\n", newMachine.Name)
+	_, err = machineClient.MachineV1alpha1().Machines().Create(newMachine)
+	if err != nil {
+		return err
+	}
+	err = wait.Poll(500*time.Millisecond, 60*time.Second, func() (bool, error) {
+		err := assureNodeCount(1, kubeClient)
+		if err == nil {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return fmt.Errorf("falied to created the new machine, err = %v", err)
```

Comment:
at this point we could grab the machine's `Status` from the  kube-api server  and put it into the logs. That would facilitate troubleshooting.